### ISSUE TITLE
How to use budgets or piggy banks for taxes?

### ISSUE THREAD

**Issue by dp87** [firefly-iii/firefly-iii#1992]
I have a question on how to best handle budgeting for taxes, because I haven't found a way that makes a lot of sense yet. The scenario is something like this: suppose you invoice a customer and you get paid for your work. As soon as you receive this payment, you know that part of it will go away in taxes, so what's the best way to use Firefly III in order to reflect this?

So far, I have thought about using budgets or piggy banks, but both of them have some limitations from what I can tell, at least when it comes to this specific scenario:

* Budgets: this should be the most reasonable option. After all, you need to _budget_ for your taxes. However, because taxes will be paid some time in the future, seeing the bar fill is basically useless, since it will be filled completely as soon as you pay your taxes. Also, in the meantime, the money still shows up in your asset accounts, so it looks like you actually can spend it while you know you shouldn't touch it
* Piggy banks: while theoretically this makes less sense than a budget, it seems to work a bit better in practice. You could create a piggy bank and then, as soon as you get paid, you transfer the correct amount to the piggy bank.

The second approach, to me, seems to work better, but it also means it would make a budget useless. Once I have used the piggy bank to track this, what good is it for me to assign the transaction with which I have paid the taxes to the "Tax" budget? Doesn't that become unnecessary at that point?

I might easily be missing something obvious here though. Do you have any tips on how to handle a situation such as this?

**Comment by JC5** [comment]
It becomes fairly unnecessary to create a budget, yes. Especially when you know exactly how much it is going to be. I would suggest using piggy banks to collect to money in, although it's not ideal (they're designed for a fixed amount).

There isn't much more you can do. Firefly III was designed for personal finance and this is a tricky fit, I'm afraid.

**Comment by dp87** [comment]
Yeah that makes sense, I figured it would be sort of a long shot for a software like Firefly III which, as you say, is for personal finance. It was worth a shot though, the software is flexible enough as it is to probably be able to work even with scenarios like this one.
